src/groupeddataframe/utils.jl

```diff
@@ -135,7 +135,11 @@ function refpool_and_array(x::AbstractArray)
     refarray = DataAPI.refarray(x)
 
     if refpool !== nothing
-        return refpool, refarray
+        if x isa PooledVector || allunique(refpool)
+            return refpool, refarray
+        else
+            return nothing, nothing
```

Comment:
@quinnj - can Arrow.jl give us an example of pooled vector with duplicates in `refpool`?

Reply:
It's technically allowed to have duplicates in the dictionary pool; but we could also call `allunique` in the `DataAPI.refpool` method in Arrow.jl to avoid this. Perhaps we should make it formally part of the `refpool` contract?

Reply:
That way we ensure other `refpool`-enabled array types get the same optimization w/o needing to hard-code PooledVector

Reply:
I don't know, I can't see duplicate values in a pool being useful at all. It's our API, so I say we enforce it. All people would have to do if they allow duplicates is call `allunique` on their refpool. I think it just comes back to the fact that all our current uses/designs for `refpool` revolve around it being a unique pool, so we might as well enforce that.